src/app/application.cpp

```diff
@@ -405,15 +405,62 @@ void Application::processParams(const QStringList &params)
         return;
     }
 #endif
+    BitTorrent::AddTorrentParams torrentParams;
+    TriStateBool skipTorrentDialog = TriStateBool::Undefined;
 
     foreach (QString param, params) {
         param = param.trimmed();
+
+        // Process strings indicating options specified by the user.
+        
+        if (param.startsWith("@savePath=")) {
+            torrentParams.savePath = param.mid(10);
+            continue;
```

Comment:
> Since it consists only of these same blocks, I do not insist on the change. In another case it could be otherwise...

I'm pretty sure in that case the current fragment would evolve into a function containing only this switch-like structure.